### DISCUSSION TITLE
NextJS api rewrite on Vercel and usage

### DISCUSSION THREAD

**Discussion by joggienl** [vercel/next.js#34393] [Help] [answered]
I have a question about using a rewrite rule in your `next.config.js` file to a graphql endpoint on a different server. For example `some-project.vercel.app/graphql` will be rewritten to `some-backend.otherdomain.com/path/or/whaterver`:

```js
// next.config.js
module.exports = {
  // ...
  async rewrites() {
    return [
      {
        source: '/graphql/:pathname*',
        destination: 'https://some-backend.otherdomain.com/path/or/whaterver'
      }
    ]
  }
};
```

When hosting this on Vercel and letting your application using this endpoint to get data, does this increase the bandwidth usage of the project? I think so because the data is essentially flowing through Vercel but I am not 100% sure and can't seem to find a good answer (or just plain looking wrong 😅 )

Hoping someone knows more on this!

**Comment by ismaelrumzan** [comment, marked as answer]
Hi there
If the response is being served from Vercel, whether we're the origin or whether we're proxying from another origin, it's still being served/handled by us. Therefore, it will be counted towards bandwidth usage.

**Reply by joggienl** [reply]
Thank you for the update here @ismaelrumzan!
